packages/core/admin/server/services/api-token.js

```diff
@@ -207,6 +266,37 @@ const regenerate = async (id) => {
   };
 };
 
+/**
+ * Refresh the expiresAt date to now + lifespan
+ *
+ * @param {string|number} id
+ *
+ * @returns {Promise<ApiToken>}
+ */
+const refresh = async (id) => {
+  const apiToken = await getById(id);
+  if (!apiToken) {
+    throw new NotFoundError('The provided token id does not exist');
+  }
+  if (!apiToken.lifespan) {
+    throw new ValidationError('Token must have lifespan to refresh');
+  }
+
+  const updatedToken = await strapi.query('admin::api-token').update({
+    select: ['id', 'accessKey'],
+    where: { id },
+    data: {
+      ...getExpirationFields(apiToken.lifespan),
+    },
+  });
+
+  if (!apiToken) {
```

Comment:
Note sure this check is really useful :sweat_smile:

Reply:
Oops, yes, it was a leftover attempt to save a query by pretending a failed update meant the token didn't exist, but then I decided against it and forgot to remove that :)